### PR TITLE
Add worker pools data source documentation

### PR DESCRIPTION
**Pull request by michieldewilde**
## Description of the change

Adds documentation for the worker pools data source.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation (non-breaking change that adds documentation)

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review

- [x]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] Pull Request is no longer marked as "draft"
- [x] Reviewers have been assigned
- [ ] Changes have been reviewed by at least one other engineer

### CODE REVIEW

**Comment by michieldewilde** [comment]
Need to base this off a different branch

File: README.md [hunk 1]
```diff
@@ -1186,3 +1187,25 @@ In addition to all arguments above, the following attributes are exported:
...
+
+#### Argument reference
+
+No arguments are supported.
+
+#### Attributes reference
+
+The following attributes are exported:
+
+- `worker_pools` - List of [`spacelift_worker_pool` data sources](#spacelift_worker_pool-data-source);
```
Comment (marcinwyszynski) [inline README.md:1209]: It's not a list of data sources but of objects with the shape identical to those returned by `spacelift_worker_pool`.

**Review by marcinwyszynski — CHANGES_REQUESTED** [review]
While at it, maybe also update the PR template, too?

File: README.md (hunk 1 above)
Comment (michieldewilde) [reply README.md:1209]: @marcinwyszynski Should I show the attributes that a `worker_pool` object could have here? Or still link it?

Comment (michieldewilde) [reply README.md:1209]: @marcinwyszynski seems to me that if I show the attributes here as well you would have to touch both of the section if the `worker_pool` data source changes. Linking seems better in that case.

**Review by marcinwyszynski — APPROVED** [review]

**Pull request merged**